src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+      addImport(compUnit, typeName);
+      typeNameMap.put(typeName, getSimpleTypeName(typeName));
+    }
+
+    CompilationUnit result = compUnit;
+    List<ImportDeclaration> imports = result.getImports();
+    for (String type : typeNameMap.keySet()) {
+      String compUnitStr = result.toString();
+
+      // Replace all instances of the fully-qualified type name with the simple type name.
+      if (compUnitStr.contains(type)) {
+        compUnitStr = compUnitStr.replace(type, typeNameMap.get(type));
+      }
+      CompilationUnit compUnitWithSimpleTypeNames;
+      try {
+        // Parse the compilation unit and set the imports.
+        compUnitWithSimpleTypeNames = JavaParser.parse(IOUtils.toInputStream(compUnitStr, "UTF-8"));
+        compUnitWithSimpleTypeNames.setImports(imports);
+      } catch (ParseException e) {
+        System.err.println("Error parsing into a compilation unit.");
+        continue;
+      } catch (IOException e) {
+        System.err.println("IOUtils error creating input stream from string.");
+        continue;
+      }
+      // Check that the simplification is correct.
+      String newFilePath = writeToFile(result, filePath, "Minimized");
+      if (checkCorrectlyMinimized(
+          newFilePath, classpath, packageName, expectedOutput, timeoutLimit)) {
+        result = compUnitWithSimpleTypeNames;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Get the simple type name of a fully qualified type name. For example, {@code java.lang.String}
+   * should be simplified to {@code String}.
+   *
+   * @param typeName fully qualified type name
+   * @return simple type name of {@code typeName}
+   */
+  public static String getSimpleTypeName(String typeName) {
+    int indexOflastSeparator = typeName.lastIndexOf('.');
+    return indexOflastSeparator > 0
+        ? typeName.substring(indexOflastSeparator + 1, typeName.length())
+        : typeName;
+  }
+
+  /**
+   * Check if a Java file has been correctly minimized. The file should not have compilation errors
+   * or runtime errors. The file should fail in the same way as the original file.
+   *
+   * @param filePath the absolute path to the Java file
+   * @param classpath classpath needed to compile/run Java file
+   * @param packageName the name of the package that the Java file is in
+   * @param expectedOutput expected output of running JUnit test suite
+   * @param timeoutLimit number of seconds allowed for the whole test suite to be run
+   * @return true if there are no compilation and no runtime errors and the output is equal to the
+   *     expected output
+   */
+  private static boolean checkCorrectlyMinimized(
+      String filePath,
+      String classpath,
+      String packageName,
+      String expectedOutput,
+      int timeoutLimit) {
+    Results res = compileAndRun(filePath, classpath, packageName, timeoutLimit);
+
+    return res.compOut.exitValue == 0
+        && res.runOut != null
+        && res.runOut.errout.isEmpty()
```

Comment:
The Minimizer seems to require that the standard output is empty.
Why does it do so?  I don't think we care about standard error when running the test, or standard output for that matter except for the JUnit errors that are output to it.
Isn't it OK for the test's output to change, so long as the failures are in the same location and with the same stack traces?
More generally, I would expect `normalizeJUnitOutput` to return a list of JUnit errors, with each one giving a full stack trace, but ignoring all output other than stack traces.
I haven't tried it myself, but you might be able to do this by running JUnit programmatically (see http://stackoverflow.com/questions/24943839/how-to-run-entire-junit-test-suite-from-within-code and other resources).